src/components/NetworkListTable.tsx

```diff
@@ -7,31 +7,41 @@ import { LxdInstance } from "types/instance";
 import { fetchNetworks } from "api/networks";
 import { isNicDevice } from "util/devices";
 import ResourceLink from "components/ResourceLink";
+import { LxdProfile } from "types/profile";
+import { useParams } from "react-router-dom";
 
 interface Props {
-  instance: LxdInstance;
   onFailure: (title: string, e: unknown) => void;
+  instance?: LxdInstance;
+  profile?: LxdProfile;
```

Comment:
Here we have the same issue, better find a generic interface that works for both cases in the same way. I think we can pass in devices here and that should work just fine. So we resolve the devices in the caller once, and actually we can reuse that for this component and the devices component.